### PR TITLE
Unit tests generation in Kotlin

### PR DESCRIPTION
**Pull request by AlexandrouR**
### What does this PR do?
Adds unit test functionality in Kotlin in the codegen module. This is required for the CLI

### Where should the reviewer start?
All changed files

### Why is it needed?
To eventually be able to generate new web3j projects in kotlin.

### CODE REVIEW

File: codegen/src/main/java/org/web3j/codegen/unit/gen/utills/MappingHelper.java [hunk 1]
```diff
@@ -21,6 +21,8 @@
 public class MappingHelper {
```
Comment (cfelde) [inline codegen/src/main/java/org/web3j/codegen/unit/gen/utills/MappingHelper.java]: Package name typo in utills

File: codegen/src/main/java/org/web3j/codegen/unit/gen/utills/ParserUtils.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
  * specific language governing permissions and limitations under the License.
  */
-package org.web3j.codegen.unit.gen;
+package org.web3j.codegen.unit.gen.utills;
```
Comment (cfelde) [inline codegen/src/main/java/org/web3j/codegen/unit/gen/utills/ParserUtils.java]: Package name typo in utills

File: codegen/src/main/java/org/web3j/codegen/unit/gen/utills/MappingHelper.java (hunk 1 above)
Comment (AlexandrouR) [reply codegen/src/main/java/org/web3j/codegen/unit/gen/utills/MappingHelper.java]: fixed

File: codegen/src/main/java/org/web3j/codegen/unit/gen/utills/ParserUtils.java (hunk 2 above)
Comment (AlexandrouR) [reply codegen/src/main/java/org/web3j/codegen/unit/gen/utills/ParserUtils.java]: fixed

File: codegen/src/main/java/org/web3j/codegen/unit/gen/Parser.java [hunk 3]
```diff
@@ -20,6 +20,7 @@
 import java.util.List;
 
 import org.junit.jupiter.api.Assertions;
+import sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl;
```
Comment (cfelde) [inline codegen/src/main/java/org/web3j/codegen/unit/gen/Parser.java]: Do we really want to use something under the sun package? It's not recommended, but be an alternative..?

File: codegen/src/test/java/org/web3j/codegen/unit/gen/Java/JavaClassGeneratorTest.java [hunk 4]
```diff
@@ -10,7 +10,7 @@
  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
  * specific language governing permissions and limitations under the License.
  */
-package org.web3j.codegen.unit.gen;
+package org.web3j.codegen.unit.gen.Java;
```
Comment (cfelde) [inline codegen/src/test/java/org/web3j/codegen/unit/gen/Java/JavaClassGeneratorTest.java]: Don't use upper case in package name

File: codegen/src/test/java/org/web3j/codegen/unit/gen/Java/JavaParserTest.java [hunk 5]
```diff
@@ -10,14 +10,16 @@
  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
  * specific language governing permissions and limitations under the License.
  */
-package org.web3j.codegen.unit.gen;
+package org.web3j.codegen.unit.gen.Java;
```
Comment (cfelde) [inline codegen/src/test/java/org/web3j/codegen/unit/gen/Java/JavaParserTest.java]: Don't use upper case in package name

File: codegen/src/test/java/org/web3j/codegen/unit/gen/Java/MethodParserTest.java [hunk 6]
```diff
@@ -10,7 +10,7 @@
  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
  * specific language governing permissions and limitations under the License.
  */
-package org.web3j.codegen.unit.gen;
+package org.web3j.codegen.unit.gen.Java;
```
Comment (cfelde) [inline codegen/src/test/java/org/web3j/codegen/unit/gen/Java/MethodParserTest.java]: Don't use upper case in package name

File: codegen/src/test/java/org/web3j/codegen/unit/gen/Java/MethodSpecGeneratorTest.java [hunk 7]
```diff
@@ -10,7 +10,7 @@
  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
  * specific language governing permissions and limitations under the License.
  */
-package org.web3j.codegen.unit.gen;
+package org.web3j.codegen.unit.gen.Java;
```
Comment (cfelde) [inline codegen/src/test/java/org/web3j/codegen/unit/gen/Java/MethodSpecGeneratorTest.java]: Don't use upper case in package name

File: codegen/src/test/java/org/web3j/codegen/unit/gen/Java/Setup.java [hunk 8]
```diff
@@ -10,7 +10,7 @@
  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
  * specific language governing permissions and limitations under the License.
  */
-package org.web3j.codegen.unit.gen;
+package org.web3j.codegen.unit.gen.Java;
```
Comment (cfelde) [inline codegen/src/test/java/org/web3j/codegen/unit/gen/Java/Setup.java]: Don't use upper case in package name

File: codegen/src/test/java/org/web3j/codegen/unit/gen/Java/JavaClassGeneratorTest.java (hunk 4 above)
Comment (AlexandrouR) [reply codegen/src/test/java/org/web3j/codegen/unit/gen/Java/JavaClassGeneratorTest.java]: That is my bad I fixed it.

File: codegen/src/main/java/org/web3j/codegen/unit/gen/Parser.java (hunk 3 above)
Comment (AlexandrouR) [reply codegen/src/main/java/org/web3j/codegen/unit/gen/Parser.java]: Fixed. It should be ParameterizedType.

**Comment by cfelde** [comment]
LGTM

File: codegen/src/main/java/org/web3j/codegen/unit/gen/kotlin/KotlinClassGenerator.java [hunk 9]
```diff
@@ -42,13 +42,25 @@ public KotlinClassGenerator(
     public void writeClass() throws Exception {
         com.squareup.kotlinpoet.ClassName EVM_ANNOTATION =
                 new com.squareup.kotlinpoet.ClassName("org.web3j", "EVMTest");
+
         com.squareup.kotlinpoet.AnnotationSpec.Builder annotationSpec =
                 com.squareup.kotlinpoet.AnnotationSpec.builder(EVM_ANNOTATION);
+
         if (JavaVersion.getJavaVersionAsDouble() < 11) {
```
Comment (cfelde) [inline codegen/src/main/java/org/web3j/codegen/unit/gen/kotlin/KotlinClassGenerator.java]: Do we really need Java 11 check for Kotlin?

**Pull request merged**